webrtc.html

```diff
@@ -509,33 +509,43 @@
 
         <ol>
           <li>
-            <p>Append an object representing the current call being handled
-            (i.e. function name and corresponding arguments) to the
-            <code>operations</code> array.</p>
+            <p>If <var>connection</var>'s <a href=
+              "#dom-peerconnection-signaling-state"><code>RTCPeerConnection</code>
+              signalingState</a> is <code>closed</code>, throw an
+              <code>InvalidStateError</code> exception and abort these steps.</p>
+          </li>
+
+          <li>
+            <p>Let <var>p</var> be the promise that the called
+            function will return.
+            </p>
+          </li>
+            <p>Add <code>p</code> to <var>operations</var>.</p>
           </li>
 
           <li>
-            <p>If the length of the <code>operations</code> array is exactly 1,
-            execute the function from the front of the queue
-            asynchronously.</p>
+            <p>If the length of <var>operations</var> is exactly 1,
+            execute the steps necessary to resolve <code>p</code>.</p>
           </li>
 
           <li>
-            <p>When the asynchronous operation completes (either successfully
-            or with an error), remove the corresponding object from the
-            <code>operations</code> array. After removal, if the array is
-            non-empty, execute the first object queued asynchronously and
-            repeat this step on completion.</p>
+            <p>When a promise in <code>operations</code> is resolved,
+            remove it from <code>operations</code>.  After removal,
+            if <code>operations</code> is non-empty, execute the steps
+            necessary to resolve the first promise
+            in <code>operations</code>.
```

Comment:
Finally! You've given me a case when all this chaining stuff you were writing might actually mean something.  Can you come up with an example where it actually matters?  Here's the best I could come up with:

pc.createOffer.then(offer => if (pc.signalingState == 'stable') pc.setLocalDescription(offer));
// Message comes in from the remote side which triggers this:
signaling.onoffer = offer => pc.setRemoteDescription(offer).then(=> pc.createAnswer(offer). then(answer => pc.setLocalDescription(answer)));

Does that set the local description first or the remote description first?  I think it should set the remote description first.  But I think according to your chaining rule, it would set the local description first.  

But I could be wrong.  The combination of your description of the chaining mixed with the rules of promising chaining make that distinction very subtle.  And that's one of the reasons I don't like PR 307.  It's so subtle and complex that unless someone implements it with exactly the setup you describe (which is rather particular to Firefox), it probably won't be correct.  And good luck to the poor javascript developer that has to understand what's supposed to happen.